plugins/modules/idrac_diagnostics.py

```diff
@@ -0,0 +1,633 @@
+          - Ignores the certificate warning while connecting to Share and is only applicable when I(share_type) is C(https).
+          - C(off) ignores the certificate warning.
+          - C(on) does not ignore the certificate warning.
+        type: str
+        choices: ["off", "on"]
+        default: "off"
+      proxy_support:
+        description:
+          - Specifies if proxy is to be used or not.
+          - C(off) does not use proxy settings.
+          - C(default_proxy) uses the default proxy settings.
+          - C(parameters_proxy) uses the specified proxy settings. I(proxy_server) is required when I(proxy_support) is C(parameters_proxy).
+          - I(proxy_support) is only applicable when I(share_type) is C(https) or C(https).
+        type: str
+        choices: ["off", "default_proxy", "parameters_proxy"]
+        default: "off"
+      proxy_type:
+        description:
+          - The proxy type of the proxy server.
+          - C(http) to select HTTP proxy.
+          - C(socks) to select SOCKS proxy.
+          - I(proxy_type) is only applicable when I(share_type) is C(https) or C(https) and when I(proxy_support) is C(parameters_proxy).
+        type: str
+        choices: [http, socks]
+        default: http
+      proxy_server:
+        description:
+          - The IP address of the proxy server.
+          - I(proxy_server) is required when I(proxy_support) is C(parameters_proxy).
+          - I(proxy_server) is only applicable when I(share_type) is C(https) or C(https) and when I(proxy_support) is C(parameters_proxy).
+        type: str
+      proxy_port:
+        description:
+          - The port of the proxy server.
+          - I(proxy_port) is only applicable when I(share_type) is C(https) or C(https) and when I(proxy_support) is C(parameters_proxy).
+        type: str
+        default: '80'
+      proxy_username:
+        description:
+          - The username of the proxy server.
+          - I(proxy_username) is only applicable when I(share_type) is C(https) or C(https) and when I(proxy_support) is C(parameters_proxy).
+        type: str
+      proxy_password:
+        description:
+          - The password of the proxy server.
+          - I(proxy_password) is only applicable when I(share_type) is C(https) or C(https) and when I(proxy_support) is C(parameters_proxy).
+        type: str
+  resource_id:
+    type: str
+    description:
+      - Id of the resource.
+      - If the value for resource ID is not provided, the module picks the first resource ID available from the list of system resources returned by the iDRAC.
+requirements:
+  - "python >= 3.9.6"
+author:
+  - "Shivam Sharma(@ShivamSh3)"
+notes:
+    - Run this module from a system that has direct access to Dell iDRAC.
+    - This module supports only iDRAC9 and above.
+    - This module supports IPv4 and IPv6 addresses.
+    - This module does not support C(check_mode).
+    - This module will always report "Changes found".
```

Comment:
Can we re-phrase "Changes found"  --> "Changes true" ?

Reply:
done